### PR TITLE
Add support for disabling "hit lighting" with osu! argon skin

### LINKED ISSUE #21824: argon skin hit lighting should toggle with setting
When it was implemented the lighting was implemented local to the judgement results, but this at very least needs to account for the user's "hit lighting" setting.

### PR DESCRIPTION
**Pull request by peppy**
- Alternative to and closes #22392.
- Closes #21824.

https://user-images.githubusercontent.com/191335/214499427-59d89d95-86c3-4094-b3ab-0a994666d53f.mp4

Same beatmap and timings as #22392 for comparison:

https://user-images.githubusercontent.com/191335/214501036-b9d8e6f7-339e-492e-a1ae-2293a4175304.mp4

Comparison:

https://user-images.githubusercontent.com/191335/214503671-13d431d4-f90b-4fe4-80f8-f0f21e229727.mp4

### CODE REVIEW

**Comment by Theighlin** [comment]
Closes #21824?

**Comment by ItsShamed** [comment]
Another video comparison for prolonged stream gameplay

https://user-images.githubusercontent.com/39100084/214578893-c1c54b27-8d54-43bc-9468-9f13de0ec438.mp4

File: osu.Game.Rulesets.Osu/Skinning/Argon/ArgonMainCirclePiece.cs [hunk 1]
```diff
@@ -176,15 +184,27 @@ private void updateStateTransforms(DrawableHitObject drawableHitObject, ArmedSta
...
+                                    .FadeColour(Color4.White, flash_in_duration * 8)
+                                    .FadeOut(flash_in_duration * 2);
+                            }
                         }
 
-                        flash.FadeTo(1, flash_in_duration, Easing.OutQuint);
+                        if (showFlash)
+                            flash.FadeTo(1, flash_in_duration, Easing.OutQuint);
+
+                        this.FadeOut(showFlash ? fade_out_time : fade_out_time / 2, Easing.OutQuad);
```
Comment (ItsShamed) [inline osu.Game.Rulesets.Osu/Skinning/Argon/ArgonMainCirclePiece.cs:206]: The thing that makes me doubt is the persistence of the *"after circle"*. This is the line that needs to be tweaked. After playing with it for a while, I've found a sweet spot with `fade_out_time / 2.75` that makes the animation the least persistent but keeps the "bounciness" of the animation.

Comment (peppy) [reply osu.Game.Rulesets.Osu/Skinning/Argon/ArgonMainCirclePiece.cs:206]: I'm not willing to go that far sorry. That is not related to hit lighting.

**Comment by bdach** [comment]
I'm just going to merge this as-is. Everyone and their grandma has an opinion about argon, so I don't think any discussion here about aesthetics is going to be productive. This PR does the job of disabling hit lighting; the rest can follow from there if we figure out how to get a clear direction out of a thousand differing subjective opinions.

**Review by bdach — APPROVED** [review]

**Pull request merged**